### ISSUE TITLE
stopped working / icon greyed out

### ISSUE THREAD

**Issue by disrupted** [sindresorhus/github-notifier-safari#10]
loved everything about this great little helper and it worked beautifully until a few days ago when it just randomly stopped working. Now the icon is always greyed out, doesn't show me notifications anymore and I can't even click it to go to the page manually.
I've tried reinstalling it without luck. Even wiped all my saved passwords in Safari and went manually through Keychain Access to delete all GitHub passwords. Then I restarted Safari and signed back in. But the issue still hasn't resolved.
Any ideas?

**Comment by Haroenv** [comment]
Interestingly for me it still works, it's greyed out when my network is disconnected.

**Comment by disrupted** [comment]
@Haroenv well, I don't really think it's a global issue with the extension. Think it might be solely on my side but I can't figure out what's wrong. Just wanted to check if there's been similar issues in the past or if anyone has hints about what might be going on.

**Comment by Jahhein** [comment]
I'm currently having this issue as well. Anyone have any ideas? I have uninstalled and reinstalled it. 

I cannot click it at all, greyed out, and no settings to fiddle with.

Using safari Version 10.1.1 (12603.2.4)

**Comment by disrupted** [comment]
sadly I never got it to work again...

**Comment by Haroenv** [comment]
I don't have any issue (although that probably doesn't help you)

**Comment by davidbonnet** [comment]
Same here, it stays grayed-out: 
![image](https://user-images.githubusercontent.com/297345/29623873-00db5008-8828-11e7-87c4-d1d9151e77c1.png)

**Comment by Morgana-Madamal** [comment]
Usually it's greyed out, if you've changed your pw or user; just log in again on git hub via Safari and it should work again. It's not a bug as far as I can see.

**Comment by Jahhein** [comment]
@Morgana-Madamal That is not at all what is going on here. I am definitely logged into Github. This has been a bug for several months now.

EDIT: Let me rephrase.. I have used this extension since a few years ago. I remember how it worked and it was never greyed out like it does now. A greyed out icon like that usually indicates a problem with a Safari extension. The extension never reports new notifications like it used to.

**Comment by schuer** [comment]
Same here: being installed on 2 machines for a long time, notifier stopped working on one machine today and is greyed out now. Tried to reinstall and logged in at GitHub again without success. Still working fine on the other machine though.

**Comment by schuer** [comment]
Console log: `Refused to execute a script because its hash, its nonce, or 'unsafe-inline' does not appear in the script-src directive of the Content Security Policy.`

Script from extension got blocked. GitHubs CSP (`script-src assets-cdn.github.com;`) does not allow for 'unsafe-inline' which shouldn’t interfere with browser extensions. Maybe it does interfere on occasion? ¯\\\_(ツ)_/¯  See https://github.com/blog/1477-content-security-policy:

> As with bookmarklets, CSP isn't supposed to interfere with any extensions either. But in reality, this isn't always the case. Specifically, in Chrome and Safari, where extensions are built in JS themselves, its typical to make modifications to the current page which may trigger a CSP exception.

**Comment by Jahhein** [comment]
Today I re-downloaded with safari 11 after install macOS High Sierra and it is now working.

**Comment by schuer** [comment]
Okay, this is crazy: GitHub Notifier on my _iMac_ was broken for several months now while on my _MacBook_ it was working fine all the time. Yesterday I cleaned up the _MacBook_ and installed a fresh High Sierra, and as a result GitHub Notifier started working on the _iMac_ (!) again! —WTF?

Me confused. But happy to have notifier working! :)

**Comment by Jahhein** [comment]
@schuer Sindre did indeed fix it and is working! Has been for a few months now.

@sindresorhus You should be in the clear to close this issue. I think.

**Comment by schuer** [comment]
Worked fine on both machines for 2 months. This week it stopped working on both machines.
Too bad, I loved the extension.
